InnerEye/ML/configs/classification/CovidHierarchicalModel.py

```diff
@@ -94,15 +99,28 @@ def should_generate_multilabel_report(self) -> bool:
 
     def get_model_train_test_dataset_splits(self, dataset_df: pd.DataFrame) -> DatasetSplits:
         if self.test_set_ids_csv:
-            test_df = pd.read_csv(self.local_dataset / self.test_set_ids_csv)
-            in_test_set = dataset_df.series.isin(test_df.series)
-            train_ids = dataset_df.series[~in_test_set].values
-            test_ids = dataset_df.series[in_test_set].values
+            test_set_ids_csv = self.local_dataset / self.test_set_ids_csv
+            test_series = pd.read_csv(test_set_ids_csv).series
+
+            all_series = dataset_df.series.values
+            check_all_test_series = all([series in all_series for series in test_series])
+            if not check_all_test_series:
+                raise ValueError(f"Not all test series from {test_set_ids_csv} were found in the dataset.")
+
+            test_set_subjects = dataset_df[dataset_df.series.isin(test_series)].subject.values
+            train_and_val_series = dataset_df[~dataset_df.subject.isin(test_set_subjects)].series.values
             num_val_samples = 400
```

Comment:
If we're making this config more generic, it could make sense to make this number some fraction of the dataset size instead of a magic number (e.g. 10% for consistency with the `else:` case)